src/payment/strategies/converge/converge-payment-strategy.spec.ts

```diff
@@ -74,144 +96,291 @@ describe('ConvergeaymentStrategy', () => {
+            jest.spyOn(paymentActionCreator, 'submitPayment')
+                .mockReturnValue(of(createErrorAction(PaymentActionType.SubmitPaymentFailed, error)));
+
+            strategy.execute(orderRequestBody);
+
+            await new Promise(resolve => process.nextTick(resolve));
+
+            expect(formPoster.postForm).toHaveBeenCalledWith('https://acs/url', {
+                PaReq: 'payer_auth_request',
+                TermUrl: 'https://callback/url',
+                MD: 'merchant_data',
+            });
+        });
+
+        it('submits order without payment data', async () => {
+            paymentMethodMock.initializationData.is3dsV2Enabled = false;
+            jest.spyOn(store.getState().paymentMethods, 'getPaymentMethodOrThrow')
+                .mockReturnValue(paymentMethodMock);
+
+            orderRequestBody = { ...orderRequestBody, payment: {...orderRequestBody.payment, methodId: 'authorizenet' } };
+
+            await strategy.execute(orderRequestBody, initializeOptions);
+
+            expect(orderActionCreator.submitOrder).toHaveBeenCalledWith(omit(orderRequestBody, 'payment'), initializeOptions);
+            expect(store.dispatch).toHaveBeenCalledWith(submitOrderAction);
+        });
+
+        it('submits payment separately', async () => {
+            await strategy.execute(orderRequestBody, initializeOptions);
+
+            expect(paymentActionCreator.submitPayment).toHaveBeenCalledWith(orderRequestBody.payment);
+            expect(store.dispatch).toHaveBeenCalledWith(submitPaymentAction);
+        });
+
+        it('returns checkout state', async () => {
+            await expect(strategy.execute(orderRequestBody)).resolves.toBe(store.getState());
+        });
+
+        it('posts 3ds data to Converge if 3ds is enabled', async () => {
+            const error = new RequestError(getResponse({
+                ...getErrorPaymentResponseBody(),
+                errors: [
+                    { code: 'three_d_secure_required' },
+                ],
+                three_ds_result: {
+                    acs_url: 'https://acs/url',
+                    callback_url: 'https://callback/url',
+                    payer_auth_request: 'payer_auth_request',
+                    merchant_data: 'merchant_data',
+                },
+                status: 'error',
+            }));
+
+            jest.spyOn(paymentActionCreator, 'submitPayment')
+                .mockReturnValue(of(createErrorAction(PaymentActionType.SubmitPaymentFailed, error)));
+
+            strategy.execute(orderRequestBody);
+
+            await new Promise(resolve => process.nextTick(resolve));
+
+            expect(formPoster.postForm).toHaveBeenCalledWith('https://acs/url', {
+                PaReq: 'payer_auth_request',
+                TermUrl: 'https://callback/url',
+                MD: 'merchant_data',
+            });
+        });
+
+        it('does not load converge if initialization options are not provided', async () => {
```

Comment:
Why did you change the name? 🤔 it was right but, it should say `if 3ds is not required`.